### ISSUE TITLE
Implement an AssetInfo PropertyEditor with ReImport Functionality

### ISSUE THREAD

**Issue by ilexp** [AdamsLair/duality#321]
#### Summary

After #285 is implemented, there needs to be a way to edit custom data attachments of a Resource in the editor and perform a re-import with the updated values. A special `PropertyEditor` can be implemented to do this and displayed when selecting a Resource where a Re-Import is possible.

#### Analysis

- Only needs to edit existing key-value pairs, no need to provide the ability to add or remove them.
  - Could just behave like editing any regular object instance.
- For the `PropertyEditor`, an invisible parent editor could host both `ResourcePropertyEditor` and the associated `AssetInfoPropertyEditor`.
  - Make sure to not break any caching mechanism and keep editor re-usage working.
- There should be a Re-Import button at the end.
  - This could actually be displayed whenever a Re-Import is possible, regardless of the availability of custom data attachments / import settings.
- There could be an Export button at the end as well, though this would be optional.
- If a re-import or export operation is performed, the `AssetInfoPropertyEditor` needs to check for new and / or removed key-value pairs.
  - Re-Import operations already [trigger editor-wide PropertyChanged events](https://github.com/AdamsLair/duality/blob/master/DualityEditor/AssetManagement/AssetManager.cs#L145), which will result in the Object Inspector doing an update.
  - Export operations don't do this. The AssetInfo editor will need to keep track of this by itself somehow. Potentially add editor infrastructure for getting notified of AssetManager operations?

**Comment by ilexp** [comment]
**Progress:**

- Did some prototyping. As it turns out, the easiest way to implement this is to go a similar route as with GameObjects, where the `GameObjectOverviewPropertyEditor` acts as a wrapper for the GameObject header and all the Component editors that are dynamically added and removed below.
- The prototype code is available [in this gist](https://gist.github.com/ilexp/203eeb9ba2f859599b1098e918e83c30).

**Comment by ilexp** [comment]
**Progress:**

- Created an [assetinfoeditor-wip](https://github.com/AdamsLair/duality/tree/assetinfoeditor-wip) feature branch for this.
- Set up the basic (overview wrapper / resource / assetinfo) class structures and connected them in a working state.
- Started implementing the AssetInfo editor side with the fixed import / export operation editor.

**Immediate ToDo:**

- Wire up AssetManager export operations to an editor-wide event and let the AssetInfo editor update itself when that happens.
- Actually retrieve key-value pairs from the AssetInfo object and display them in the AssetInfo editor.
- Maybe some kind of feedback when pressind Import / Export and the operation succeeds?

**Comment by ilexp** [comment]
**Progress:**

- Wired up export events.
- Can now edit key-value pairs from custom AssetInfo data.
- Added system sound when clicking Import / Export and successfully performing the operation.
- Added help system integration.
- Did a lot of documentation.

**Immediate ToDo:**

- Merge this into the master branch and do more testing.

**Comment by ilexp** [comment]
Merged into master, awaiting further testing before binary release. Closing this.
